ktor-client/ktor-client-jetty/jvm/src/io/ktor/client/engine/jetty/JettyHttp2Engine.kt

```diff
@@ -20,27 +22,49 @@ internal class JettyHttp2Engine(override val config: JettyEngineConfig) : HttpCl
         )
     }
 
-    private val jettyClient = HTTP2Client().apply {
-        addBean(config.sslContextFactory)
-        check(config.proxy == null) { "Proxy unsupported in Jetty engine." }
+    override val supportedCapabilities = setOf(HttpTimeout)
 
-        executor = QueuedThreadPool().apply {
-            name = "ktor-jetty-client-qtp"
-        }
-
-        start()
-    }
+    /**
+     * Cache that keeps least recently used [HTTP2Client] instances. Set "0" to avoid caching.
+     */
+    private val clientCache = createLRUCache(::createJettyClient, HTTP2Client::stop, config.clientCacheSize)
 
     override suspend fun execute(data: HttpRequestData): HttpResponseData {
         val callContext = callContext()
+        val jettyClient =
+            clientCache[data.getCapabilityOrNull(HttpTimeout)]
+                ?: error("Http2Client can't be constructed")
```

Comment:
Same issue here: error message is not descriptive enough.

Reply:
I updated the message.